clang/lib/Driver/ToolChains/CommonArgs.cpp

```diff
@@ -166,7 +155,7 @@ static bool useFramePointerForTargetByDefault(const llvm::opt::ArgList &Args,
 
 static bool useLeafFramePointerForTargetByDefault(const llvm::Triple &Triple) {
   if (Triple.isAArch64() || Triple.isPS() || Triple.isVE() ||
-      (Triple.isAndroid() && Triple.isRISCV64()))
+      (Triple.isAndroid() && !Triple.isARM()))
```

Comment:
I didn't intentionally remove frame pointer for leaf functions for thumb in Android. But I also don't have reason to add it back:
1. On Android, we can't rely on stack frame pointer based unwinding for arm32 binaries. Because arm and thumb use different registers as frame pointers.
2. On other architectures that we want to support stack frame pointer based unwinding (aarch64, riscv64, x86_64), we don't need frame pointer for leaf functions.

Reply:
In https://godbolt.org/z/Krc3eq46h, I tried generating thumb instruction with and without -mno-omit-leaf-frame-pointer. They generate the same instructions not setting up frame pointers for leaf functions.